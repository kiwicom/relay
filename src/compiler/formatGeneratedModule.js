// @flow

/*::

type FormatModuleInput = $ReadOnly<{|
  documentType: $FlowFixMe,
  concreteText: string,
  typeText: string,
  sourceHash: string,
  docText?: string,
  hash?: string,
  devOnlyAssignments?: string,
|}>;

*/

module.exports = function formatGeneratedModule(
  {
    documentType,
    docText,
    concreteText,
    typeText,
    hash,
    sourceHash,
    devOnlyAssignments,
  } /*: FormatModuleInput */,
) {
  const documentTypeImport = documentType
    ? `import type { ${documentType} } from 'relay-runtime';`
    : '';
  const docTextComment = docText ? '\n/*\n' + docText.trim() + '\n*/\n' : '';
  const hashText = hash ? `\n * ${hash}` : '';
  const devOnlyAddons = devOnlyAssignments
    ? `\nif (__DEV__) {\n  ${devOnlyAssignments}\n}`
    : '';

  return `/**
 * ${'@'}flow${hashText}
 */

/* eslint-disable */

'use strict';

/*::
${documentTypeImport}
${typeText || ''}
*/

${docTextComment}
const node/*: ${documentType || 'empty'}*/ = ${concreteText};
// prettier-ignore
(node/*: any*/).hash = '${sourceHash}';${devOnlyAddons}
module.exports = node;
`;
};
