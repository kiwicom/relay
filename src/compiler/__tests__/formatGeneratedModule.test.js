// @flow

import formatGeneratedModule from '../formatGeneratedModule';

it('renders minimal input correctly', () => {
  expect(
    formatGeneratedModule({
      documentType: 'DOCUMENT_TYPE',
      concreteText: 'CONCRETE_TEXT',
      typeText: 'TYPE_TEXT',
      sourceHash: 'SOURCE_HASH',
      devOnlyAssignments: '',
    }),
  ).toMatchSnapshot();
});

it('renders full input correctly', () => {
  expect(
    formatGeneratedModule({
      documentType: 'DOCUMENT_TYPE',
      concreteText: 'CONCRETE_TEXT',
      typeText: 'TYPE_TEXT',
      sourceHash: 'SOURCE_HASH',
      docText: 'DOC_TEXT',
      hash: 'HASH',
      devOnlyAssignments: 'DEV_ONLY_ASSIGNMENTS',
    }),
  ).toMatchSnapshot();
});
