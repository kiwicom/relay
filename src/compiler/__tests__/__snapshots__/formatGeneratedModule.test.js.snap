// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders full input correctly 1`] = `
"/**
 * @flow
 * HASH
 */

/* eslint-disable */

'use strict';

/*::
import type { DOCUMENT_TYPE } from 'relay-runtime';
TYPE_TEXT
*/


/*
DOC_TEXT
*/

const node/*: DOCUMENT_TYPE*/ = CONCRETE_TEXT;
// prettier-ignore
(node/*: any*/).hash = 'SOURCE_HASH';
/*:: declare var __DEV__: boolean; */
if (__DEV__) {
  DEV_ONLY_ASSIGNMENTS
}
module.exports = node;
"
`;

exports[`renders minimal input correctly 1`] = `
"/**
 * @flow
 */

/* eslint-disable */

'use strict';

/*::
import type { DOCUMENT_TYPE } from 'relay-runtime';
TYPE_TEXT
*/


const node/*: DOCUMENT_TYPE*/ = CONCRETE_TEXT;
// prettier-ignore
(node/*: any*/).hash = 'SOURCE_HASH';
module.exports = node;
"
`;
